src/js/ui/user/variantperf/variantPerfView.js

```diff
@@ -14,140 +17,141 @@ export default function view(vnode) {
-            <td class="variantPerfMemo"> {data.stat.count.berserk} </td>
-            <td class="variantPerfMemo"> {Math.round((data.stat.count.berserk / data.stat.count.all) * 100) + '%'} </td>
-          </tr>
-          <tr>
-            <th class="variantPerfMemo"> Disconnect </th>
-            <td class="variantPerfMemo" > {data.stat.count.disconnects} </td>
-            <td class="variantPerfMemo"> {Math.round((data.stat.count.disconnects / data.stat.count.all) * 100) + '%'} </td>
-          </tr>
-          <tr> <td class="variantPerfSpacer" colspan="3"> </td> </tr>
-          <tr>
-            <th class="variantPerfHeading" colspan="3"> Streaks </th>
-          </tr>
-          <tr class={isEmpty(data.stat.playStreak)}>
-            <th class="variantPerf"> Playing streak </th>
-            <td class="variantPerf"> {data.stat.playStreak.nb.max.v} </td>
-            <td class="variantPerf"> {data.stat.playStreak.lastDate.substring(0, 10)} </td>
-          </tr>
-          <tr class={isEmpty(data.stat.resultStreak.win.max.v)}>
-            <th class="variantPerf"> {i18n('winStreak')} </th>
-            <td class="variantPerf"> <span class="progress positive"> {data.stat.resultStreak.win.max.v ? data.stat.resultStreak.win.max.v : ''} </span> </td>
-            <td class="variantPerf"> <span class="progress positive"> {data.stat.resultStreak.win.max.to ? data.stat.resultStreak.win.max.to.at.substring(0, 10) : ''} </span> </td>
-          </tr>
-          <tr class={isEmpty(data.stat.resultStreak.loss.max.v)}>
-            <th class="variantPerf"> Losing streak </th>
-            <td class="variantPerf"> <span class="progress negative"> {data.stat.resultStreak.loss.max.v ? data.stat.resultStreak.loss.max.v : ''} </span> </td>
-            <td class="variantPerf"> <span class="progress negative"> {data.stat.resultStreak.loss.max.to ? data.stat.resultStreak.loss.max.to.at.substring(0, 10) : ''} </span> </td>
-          </tr>
-          <tr> <td class="variantPerfSpacer" colspan="3"> </td> </tr>
-          </tbody>
-        </table>
-        <div class={'variantPerfGames noPadding ' + isEmpty(data.stat.bestWins.results.length)}>
-          <div class="variantPerfHeading"> Best Wins </div>
-          {data.stat.bestWins.results.map(p => renderGame(p))}
-        </div>
-      </div>
-    );
-  }
+  return layout.free(header, () => renderBody(ctrl));
+}
 
-  return layout.free(header, renderBody);
+function renderBody(ctrl) {
+  const data = ctrl.variantPerfData();
+
+  if (!data) return null;
+
+  const days = Math.floor(data.stat.count.seconds / (60 * 60 * 24));
+  const hours = Math.floor(data.stat.count.seconds / (60 * 60)) - days * 24;
+  const mins = Math.floor(data.stat.count.seconds / 60) - days * 24 * 60 - hours * 60;
+
+  return (
+    <div class="variantPerfBody native_scroller page">
+      <canvas oncreate={node => createGraph(data, node)} onupdate={node => updateGraph(data, node)} />
```

Comment:
I think you can just add a `{ key: isPortrait ? 'portrait' : 'landscape' }` to force canvas recreation on orientation change, and avoid the whole `onupdate` logic.